### PR TITLE
[CLOUDTRUST-5142] Add events when accreditations are granted/revoked

### PR DESCRIPTION
**Pull request by fperot74**
Add events when an accreditation is granted/revoked
Rename credentialID detail to credential_id in events for consistency reason (credential_id is used in our Keycloak modules)

### CODE REVIEW

File: internal/keycloakb/accreditationsmodule.go [hunk 1]
```diff
@@ -53,9 +63,9 @@ func RevokeAccreditations(kcUser *kc.UserRepresentation) bool {
 // AccreditationsProcessor interface
 type AccreditationsProcessor interface {
 	HasActiveAccreditations() bool
-	AddAccreditation(creationDate time.Time, name string, validity string)
-	RevokeAll() bool
-	RevokeTypes(accreditationsTypes []string) bool
+	AddAccreditation(creationDate time.Time, name string, validity string) AccreditationRepresentation
+	RevokeAll(notif func(AccreditationRepresentation)) bool
+	RevokeTypes(accreditationsTypes []string, notif func(AccreditationRepresentation)) bool
```
Comment (basbeu) [inline internal/keycloakb/accreditationsmodule.go:68]: why returning a bool when it is never used ?

File: internal/keycloakb/accreditationsmodule.go [hunk 2]
```diff
@@ -117,19 +127,21 @@ func (ap *accredsProcessor) AddAccreditation(creationDate time.Time, name string
...
-		if ap.RevokeType(k) {
+		if ap.RevokeType(k, notif) {
 			res = true
 		}
 	}
 	return res
 }
 
-func (ap *accredsProcessor) RevokeType(accreditationsType string) bool {
+func (ap *accredsProcessor) RevokeType(accreditationsType string, notif func(AccreditationRepresentation)) bool {
```
Comment (basbeu) [inline internal/keycloakb/accreditationsmodule.go:144]: should not be a private method ?

Comment (basbeu) [reply internal/keycloakb/accreditationsmodule.go:144]: or exposed in the interface ?

File: pkg/validation/component.go [hunk 3]
```diff
@@ -60,27 +60,33 @@ type Component interface {
...
-	originEvent         string
+	keycloakClient       KeycloakClient
+	tokenProvider        TokenProvider
+	archiveDBModule      ArchiveDBModule
+	eventReporterModule  EventsReporterModule
+	accredsService       AccreditationsServiceClient
+	configDBModule       ConfigurationDBModule
+	logger               log.Logger
+	originEvent          string
+	unknownAgentRealm    string
```
Comment (basbeu) [inline pkg/validation/component.go:71]: Why not to use constants instead of field of a struct ?

Comment (fperot74) [reply pkg/validation/component.go:71]: I just copied what was done for unknownAgentUserID and unknownAgentUsername... You ask yourself why you did that for those 2 fields! ;-)

**Review by basbeu — APPROVED** [review]

**Pull request merged**